src/storage/raw/raw_mvcc.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+// #[PerformanceCriticalPath]
+use crate::storage::kv::{Iterator, Result, Snapshot};
+
+use engine_traits::{CfName, DATA_KEY_PREFIX_LEN};
+use engine_traits::{IterOptions, ReadOptions};
+use txn_types::{Key, TimeStamp, Value};
+
+#[derive(Clone)]
+pub struct RawMvccSnapshot<S: Snapshot> {
+    snap: S,
+}
+
+impl<S: Snapshot> RawMvccSnapshot<S> {
+    pub fn from_snapshot(snap: S) -> Self {
+        RawMvccSnapshot { snap }
+    }
+
+    pub fn seek_first_key_value_cf(
+        &self,
+        cf: Option<CfName>,
+        opts: Option<ReadOptions>,
+        key: &Key,
+    ) -> Result<Option<Value>> {
+        let mut iter_opt = IterOptions::default();
+        iter_opt.set_fill_cache(opts.map_or(true, |v| v.fill_cache()));
+        iter_opt.use_prefix_seek();
+        iter_opt.set_prefix_same_as_start(true);
+        let end_key = key.clone().append_ts(TimeStamp::zero());
+        iter_opt.set_upper_bound(end_key.as_encoded(), DATA_KEY_PREFIX_LEN);
+        let mut iter = match cf {
+            Some(cf_name) => self.iter_cf(cf_name, iter_opt)?,
+            None => self.iter(iter_opt)?,
+        };
+        if iter.seek(key)? {
+            // Following line can be removed after mce padding in api v2.
```

Comment:
Actually, without padding, this function will not work as expected, returning `None` is also wrong.

Reply:
+1

Reply:
emm... yes. So currently, case of apiv2 are using keys with same length. I think we can add more cases with different length keys after do mce padding in raw_put/raw_batch_put functions. Mce padding PR is one the way by @pingyu.

Reply:
For now, i will remove this equal judgement.